### PR TITLE
Make some const prop mir-opt tests `unit-test`s

### PR DESCRIPTION
**Pull request by Noratrieb**
Most of these have no or only tiny diffs beyond line numbers being changed (would it make sense to not have line numbers in mir-opt tests?). Some things changed a bit, but I think it should all be fine, not sure though.

### CODE REVIEW

**Comment by JakobDegen** [comment]
> would it make sense to not have line numbers in mir-opt tests?

This has been discussed a number of times, and people have mixed opinions on it. I still think we should do it though. I think there was a Zulip thread about this somewhere.

I'll review later as well, but no perms so Mark will need to review himself or hand off to mir-opt or something

**Comment by JakobDegen** [comment]
I'm going to wait to review this until after #99780 lands, for obvious reasons

File: src/test/mir-opt/const_prop/bad_op_div_by_zero.main.ConstProp.diff [hunk 1]
```diff
@@ -24,10 +24,9 @@
           StorageLive(_3);                 // scope 1 at $DIR/bad_op_div_by_zero.rs:+2:18: +2:19
 -         _3 = _1;                         // scope 1 at $DIR/bad_op_div_by_zero.rs:+2:18: +2:19
 -         _4 = Eq(_3, const 0_i32);        // scope 1 at $DIR/bad_op_div_by_zero.rs:+2:14: +2:19
--         assert(!move _4, "attempt to divide `{}` by zero", const 1_i32) -> bb1; // scope 1 at $DIR/bad_op_div_by_zero.rs:+2:14: +2:19
 +         _3 = const 0_i32;                // scope 1 at $DIR/bad_op_div_by_zero.rs:+2:18: +2:19
 +         _4 = const true;                 // scope 1 at $DIR/bad_op_div_by_zero.rs:+2:14: +2:19
-+         assert(!const true, "attempt to divide `{}` by zero", const 1_i32) -> bb1; // scope 1 at $DIR/bad_op_div_by_zero.rs:+2:14: +2:19
```
Comment (JakobDegen) [inline src/test/mir-opt/const_prop/bad_op_div_by_zero.main.ConstProp.diff:30]: This is one example but probably non-critical

File: src/test/mir-opt/const_prop/boxes.main.ConstProp.diff [hunk 2]
```diff
@@ -26,12 +26,11 @@
           StorageLive(_3);                 // scope 0 at $DIR/boxes.rs:+1:14: +1:22
 -         _4 = SizeOf(i32);                // scope 2 at $DIR/boxes.rs:+1:14: +1:22
 -         _5 = AlignOf(i32);               // scope 2 at $DIR/boxes.rs:+1:14: +1:22
--         _6 = alloc::alloc::exchange_malloc(move _4, move _5) -> bb1; // scope 2 at $DIR/boxes.rs:+1:14: +1:22
 +         _4 = const 4_usize;              // scope 2 at $DIR/boxes.rs:+1:14: +1:22
 +         _5 = const 4_usize;              // scope 2 at $DIR/boxes.rs:+1:14: +1:22
-+         _6 = alloc::alloc::exchange_malloc(const 4_usize, const 4_usize) -> bb1; // scope 2 at $DIR/boxes.rs:+1:14: +1:22
```
Comment (JakobDegen) [inline src/test/mir-opt/const_prop/boxes.main.ConstProp.diff:32]: Also non-critical here

File: src/test/mir-opt/const_prop/invalid_constant.main.ConstProp.diff [hunk 3]
```diff
@@ -36,37 +36,28 @@
...
--         _4 = (_5.1: E);                  // scope 4 at $DIR/invalid_constant.rs:+13:34: +13:57
--         _3 = [move _4];                  // scope 1 at $DIR/invalid_constant.rs:+13:24: +13:60
-+         _4 = const Scalar(0x00000004): E; // scope 4 at $DIR/invalid_constant.rs:+13:34: +13:57
-+                                          // mir::Constant
-+                                          // + span: $DIR/invalid_constant.rs:28:34: 28:57
-+                                          // + literal: Const { ty: E, val: Value(Scalar(0x00000004)) }
-+         _3 = [const Scalar(0x00000004): E]; // scope 1 at $DIR/invalid_constant.rs:+13:24: +13:60
-+                                          // mir::Constant
-+                                          // + span: $DIR/invalid_constant.rs:28:24: 28:60
-+                                          // + literal: Const { ty: E, val: Value(Scalar(0x00000004)) }
```
Comment (JakobDegen) [inline src/test/mir-opt/const_prop/invalid_constant.main.ConstProp.diff:56]: This regression seems problematic though. Can you remove the `// unit-test` here for now and mention this test in the follow-up issue?

**Review by JakobDegen — CHANGES_REQUESTED** [review]
Overall looks good. Unfortunately, const prop currently uses the mir-opt-level to determine how much constant propagation to do, which plays poorly with `// unit-test` since that emits `-Z mir-opt-level=0`. This causes a couple of regressions (see below) in places where const prop is now less aggressive because of the flag. Imo it's a bad idea to have the same flag control both whether other optimizations run and how aggressive const prop is. I think we should instead have `-Z mir-const-prop-level` which defaults to the same value as `-Z mir-opt-level` but can be set separately. Assuming Oli agrees with this assessment, after this is merged can you file a follow-up issue to that effect?

**Edit:** I hit this in an unrelated case, so I've gone and [opened a Zulip thread about it](https://rust-lang.zulipchat.com/#narrow/stream/189540-t-compiler.2Fwg-mir-opt/topic/.60mir_opt_level.60.20overloaded.3F)

File: src/test/mir-opt/const_prop/invalid_constant.main.ConstProp.diff (hunk 3 above)
Comment (Nilstrieb) [reply src/test/mir-opt/const_prop/invalid_constant.main.ConstProp.diff]: Removed it

**Comment by Nilstrieb** [comment]
I forgot about this PR :D, but fixed the one bad case now.

**Comment by Nilstrieb** [comment]
Let's reroll to get it through
r? rust-lang/mir-opt

**Comment by Nilstrieb** [comment]
r? rust-lang/mir-opt

**Comment by Nilstrieb** [comment]
it does not want to do the thing :/

**Comment by JakobDegen** [comment]
Lets just

r? @wesleywiser 

since he reviewed a similar PR a bit ago

**Pull request merged**